stdlib/public/core/Collection.swift

```diff
@@ -965,30 +960,34 @@ extension Collection {
 
   /// Returns the distance between two indices.
   ///
-  /// Unless the collection conforms to the `BidirectionalCollection` protocol,
-  /// `start` must be less than or equal to `end`.
-  ///
   /// - Parameters:
   ///   - start: A valid index of the collection.
   ///   - end: Another valid index of the collection. If `end` is equal to
   ///     `start`, the result is zero.
-  /// - Returns: The distance between `start` and `end`. The result can be
-  ///   negative only if the collection conforms to the
-  ///   `BidirectionalCollection` protocol.
+  /// - Returns: The distance between `start` and `end`.
   ///
   /// - Complexity: O(1) if the collection conforms to
   ///   `RandomAccessCollection`; otherwise, O(*n*), where *n* is the
   ///   resulting distance.
   @_inlineable
   public func distance(from start: Index, to end: Index) -> IndexDistance {
-    _precondition(start <= end,
-      "Only BidirectionalCollections can have end come before start")
-
-    var start = start
+    var _start: Index
+    let _end: Index
+    let step: IndexDistance
+    if start > end {
```

Comment:
You can use `>` on indexes?

Reply:
Yep! `Comparable` is their one requirement.

Reply:
Huh, somehow I had it in my mind that they were just Equatable. Well this is nice, then.

Reply:
You're not wrong! That was the case before the big collections-move-indices change: http://swiftdoc.org/v2.2/protocol/ForwardIndexType/